nexus/src/authz/api_resources.rs

```diff
@@ -321,6 +331,24 @@ authz_resource! {
     polar_snippet = Custom,
 }
 
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    DeserializeFromStr,
+    Display,
+    Eq,
+    FromStr,
+    PartialEq,
+    SerializeDisplay,
+    JsonSchema,
+)]
+#[display(style = "kebab-case")]
+pub enum OrganizationRoles {
```

Comment:
Yeah, that's fair.  I considered making this an argument to the `authz` macro instead.  I'd basically change `roles_allowed = true` to `allowed_roles = <an enum type>`.

Alternatively, I could remove the reference to `{Foo}Roles` from the macro altogether -- it'd be pretty reasonable to just impl the `ApiResourceWithRolesType` trait explicitly for the few types where we want it.

Reply:
I did the latter in a447ae61.  I like this better -- it also means I didn't have to generate unused enums for Silos and Projects (those will come in a follow-on PR, but I previously had them in this PR because the macro did this).  Thanks for raising this.